packages/backend/src/tokens/request.ts

```diff
@@ -148,10 +150,35 @@ function assertProxyUrlOrDomain(proxyUrlOrDomain: string | undefined) {
 }
 
 export async function authenticateRequest(options: AuthenticateRequestOptions): Promise<RequestState> {
-  options.frontendApi = parsePublishableKey(options.publishableKey)?.frontendApi || options.frontendApi || '';
-  options.apiUrl = options.apiUrl || API_URL;
-  options.apiVersion = options.apiVersion || API_VERSION;
-  options.headerToken = options.headerToken?.replace('Bearer ', '');
+  const { request: isomorphicRequest } = options;
+  const isomorphicRequestCookies = isomorphicRequest ? parseIsomorphicRequestCookies(isomorphicRequest) : undefined;
+  const isomorphicRequestSearchParams = isomorphicRequest?.url
+    ? new URL(isomorphicRequest.url)?.searchParams
+    : undefined;
+
+  options = {
+    ...options,
+    frontendApi: parsePublishableKey(options.publishableKey)?.frontendApi || options.frontendApi,
+    apiUrl: options.apiUrl || API_URL,
+    apiVersion: options.apiVersion || API_VERSION,
+    headerToken:
+      stripAuthorizationHeader(options.headerToken) ||
+      stripAuthorizationHeader(isomorphicRequest?.headers?.get(constants.Headers.Authorization)) ||
+      undefined,
+    cookieToken: options.cookieToken || isomorphicRequestCookies?.(constants.Cookies.Session) || undefined,
+    clientUat: options.clientUat || isomorphicRequestCookies?.(constants.Cookies.ClientUat) || undefined,
+    origin: options.origin || isomorphicRequest?.headers?.get(constants.Headers.Origin) || undefined,
+    host: options.host || isomorphicRequest?.headers?.get(constants.Headers.Host) || undefined,
+    forwardedHost:
+      options.forwardedHost || isomorphicRequest?.headers?.get(constants.Headers.ForwardedHost) || undefined,
+    forwardedPort:
+      options.forwardedPort || isomorphicRequest?.headers?.get(constants.Headers.ForwardedPort) || undefined,
+    forwardedProto:
+      options.forwardedProto || isomorphicRequest?.headers?.get(constants.Headers.ForwardedProto) || undefined,
+    referrer: options.referrer || isomorphicRequest?.headers?.get(constants.Headers.Referrer) || undefined,
+    userAgent: options.userAgent || isomorphicRequest?.headers?.get(constants.Headers.UserAgent) || undefined,
```

Comment:
🙃 could we create a helper method for the repetitive parts to reduce the length of each size and make it more readable?
Example:

```typescript
const getFromReq = (req, key) => req?.headers?.get(key) || undefined ;

// usage
options = {
// ...
forwardedProto: options.forwardedProto  || getFromReq(isomorphicRequest, constants.Headers.ForwardedProto)
//...
}
```